client/state/reader/posts/normalization-rules.js

```diff
@@ -107,6 +107,30 @@ export function classifyPost( post ) {
 	return post;
 }
 
+/**
+ * Add discover properties to a post
+ * @param  {Object} post - the post to extend
+ * @return {Object}      - the post with discover properties
+ */
+
+export function addDiscoverProperties( post ) {
+	const isDiscover = !! ( get( post, 'discover_metadata' ) || DISCOVER_BLOG_ID === get( post, 'site_ID' ) );
+	let discoverFormat;
+
+	if ( isDiscover ) {
+		const formats = get( post, 'discover_metadata.discover_fp_post_formats' );
+		const pickFormat = find( formats, format => format.slug !== 'pick' );
+
+		// if there is no pick format the post is a discover feature
+		discoverFormat = pickFormat ? pickFormat.slug : 'feature';
+	}
+
+	post.is_discover = isDiscover;
+	post.discover_format = discoverFormat;
```

Comment:
I had a similar thought. I tend to lean more toward knowing all that _could_ be on the object.  That way properties don't suddenly appear under specific conditions.

I'm fully open to changing however.